markets/perps-market/contracts/storage/AsyncOrder.sol

```diff
@@ -260,16 +260,20 @@ library AsyncOrder {
         // as a maker (reducing skew) as it's now taking (increasing skew) in the opposite direction. hence,
         // a different fee is applied on the proportion increasing the skew.
 
-        // proportion of size that's on the other direction
-        uint takerSize = MathUtil.abs((marketSkew + sizeDelta).divDecimal(sizeDelta));
-        uint makerSize = DecimalMath.UNIT - takerSize;
-        uint takerFee = MathUtil.abs(notionalDiff).mulDecimal(takerSize).mulDecimal(
-            orderFeeData.takerFee
-        );
-        uint makerFee = MathUtil.abs(notionalDiff).mulDecimal(makerSize).mulDecimal(
+        // The proportions are computed as follows:
+        // makerSize = abs(marketSkew) => since we are reversing the skew, the maker size is the current skew
+        // takerSize = abs(marketSkew + sizeDelta) => since we are reversing the skew, the taker size is the new skew
+        //
+        // we then multiply the sizes by the fill price to get the notional value of each side, and that times the fee rate for each side
+
+        uint makerFee = MathUtil.abs(marketSkew).mulDecimal(fillPrice).mulDecimal(
             orderFeeData.makerFee
         );
 
```

Comment:
Changed how we calculate the maker and taker sides since previous version had a rounding error at some proportions. i.e. with 
```
sizeDelta = -3 marketSkew
that means makerSide is 1/3 and takerSide is 2/3 that rounded at 
makerSide = 333333333333333333
takerSide = 666666666666666667 
```